benchmark_runner/common/clouds/shared/s3/s3_operations.py

```diff
@@ -0,0 +1,233 @@
+
+
+import os
+import boto3
+import typeguard
+from botocore.exceptions import ClientError
+from os import listdir
+from os.path import isfile, join
+
+from benchmark_runner.common.clouds.shared.s3.s3_operations_exceptions import S3FileNotUploaded, S3FileNotDownloaded, S3FileNotDeleted, S3KeyNotCreated
+from benchmark_runner.main.environment_variables import environment_variables
+
+
+class S3Operations:
+    """ This class is responsible for S3 operations """
+
+    def __init__(self, region_name: str = ''):
+        # environment variables
+        self.__environment_variables_dict = environment_variables.environment_variables_dict
+        # must add region for pytest
+        if region_name:
+            self.__region = region_name
+        else:
+            self.__region = self.__environment_variables_dict.get('region_name', '')
+        # must be None for pytest
+        self.__endpoint_url = self.__environment_variables_dict.get('endpoint_url', None)
+        self.__aws_access_key_id = self.__environment_variables_dict.get('access_key_id', '')
+        self.__aws_secret_access_key = self.__environment_variables_dict.get('secret_access_key', '')
+        self.__s3_client = boto3.client(service_name='s3',
+                                        region_name=self.__region,
+                                        endpoint_url=self.__endpoint_url,
+                                        aws_access_key_id=self.__aws_access_key_id,
+                                        aws_secret_access_key=self.__aws_secret_access_key)
+
+    @typeguard.typechecked
+    def upload_file(self, file_name_path: str, bucket: str, key: str, upload_file: str):
+        """
+        This method upload file to s3
+        :param file_name_path:'/home/user/test.txt'
+        :param bucket:'devops-ais'
+        :param key:'test-data'
+        :param upload_file:'test.txt'
+        :return:
+        """
+        try:
+            self.__s3_client.upload_file(Filename=file_name_path,
+                                         Bucket=bucket,
+                                         Key=f'{key}/{upload_file}',
+                                         ExtraArgs={'ServerSideEncryption': 'AES256'})
+
+        except ClientError:
+            raise
```

Comment:
Missing exception?

Reply:
We dont want to catch it, we just move it forward to get more error details